### PR TITLE
Implement beatmap set information overlay

### PR DESCRIPTION
**Pull request by DrabWeb**
screen shot 2017-09-12 at 10 09 41 pm

API integration coming in a follow up PR.

### CODE REVIEW

**Comment by peppy** [comment]
A few visual issues I noticed:

- Would be nice if the stats tween like they do on the BeatmapDetails display, rather than starting from zero (https://streamable.com/9q39c)
- Success rate appears from wrong side when loading a new beatmap (https://streamable.com/0rvn2)
- Appear animation for video/novideo buttons feels a bit weird (https://streamable.com/g1xb2)
- Weird button effect when loading a new beatmap (https://streamable.com/dbh0n)

**Comment by DrabWeb** [comment]
https://streamable.com/mpbxz

File: osu.Game/Overlays/OnlineBeatmapSet/PreviewButton.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        {
+            bg.FadeColour(Color4.Black.Opacity(0.25f), 100);
+            base.OnHoverLost(state);
+        }
+
+        private void loadPreview()
+        {
+            if (preview == null || preview.HasCompleted && BeatmapSet != null)
+            {
+                preview = audio.Track.Get(BeatmapSet.OnlineInfo.Preview);
```
Comment (peppy) [inline osu.Game/Overlays/OnlineBeatmapSet/PreviewButton.cs]: this needs to be done asynchronously. currently it causes hard blocking on the update thread.

i'd possibly consider only loading it when the play button is hit.

**Review by peppy — CHANGES_REQUESTED** [review]
Fix synchronous preview loading logic

**Comment by peppy** [comment]
Why is there a `User` inside `BeatmapSetOnlineInfo`? I'm pretty sure this isn't the correct place for it.

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
Would like to offer a bounty of $500 for this feature. Please reach out via email or discord DM to claim.

**Pull request merged**